### PR TITLE
feat(typescript-estree): check modifiers on every node

### PR DESCRIPTION
**Pull request by fisker**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

I copied those checks from [Prettier repo](https://github.com/prettier/prettier/blob/e0df0f37ad3cfd87dcb739cd734d62052f4aab11/src/language-js/parse/postprocess/typescript.js#L63), they are originally copied from typescript package.

### CODE REVIEW

**Review by bradzacher — APPROVED** [review]
LGTM! THanks!

File: packages/typescript-estree/src/convert.ts [hunk 1]
```diff
@@ -3134,13 +3095,195 @@ export class Converter {
...
+            modifier.kind,
+          )}' modifier cannot appear on a parameter.`,
+        );
+      }
+
+      // `checkParameter` function in `typescript`
+      if (
+        node.kind === SyntaxKind.Parameter &&
+        // @ts-expect-error -- internal?
+        ts.hasSyntacticModifier?.(
```
Comment (fisker) [inline packages/typescript-estree/src/convert.ts:3270]: I'm not sure what to do with this, so I added `?.` to be safe.

@bradzacher Better idea?

Comment (bradzacher) [reply packages/typescript-estree/src/convert.ts:3270]: missed this during the review...

what does this function do? I haven't seen it before

Comment (fisker) [reply packages/typescript-estree/src/convert.ts:3270]: I don't know, copied from typescript package.

Comment (bradzacher) [reply packages/typescript-estree/src/convert.ts:3270]: I would just replace this with this to remove the dependency on internals
```
modifier.kind === SyntaxKind.PublicKeyword ||
modifier.kind === SyntaxKind.PrivateKeyword ||
modifier.kind === SyntaxKind.ProtectedKeyword ||
modifier.kind === SyntaxKind.ReadonlyKeyword
```

**Review by bradzacher — APPROVED** [review]

**Pull request merged**